OOP_Lesson5/Task1.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Сделайте класс User c полями name, age. Добавьте методы setAge, addAge(добавить возраст пользователю), 
+ * isAgeCorrect(проверить возраст от 18 включительно до 60 включительно). Сделайте поля класса приватными.
+ * Попробуйте вызвать метод isAgeCorrect снаружи класса. Убедитесь, что это будет вызывать ошибку. 
+ */
+/**
+ * [Description User]
+ */
+class User
+{
+    private string $name;
+    private int $age;
+    /**
+     * @param string $name
+     * @param int $age
+     */
+    public function __construct(string $name, int $age) 
+    {
+        $this->name = $name;
+        $this->age = $age;
+    }
+    /**
+     * @return string
+     */
+    public function getName() : string 
+    {
+        return $this->name;
+    }
+    /**
+     * @return int
+     */
+    public function getAge() : int 
+    {
+        return $this->age;
+    }
+    /**
+     * @param int $age
+     * 
+     * @return bool
+     */
+    private function isAgeCorrect(int $age) : bool 
+    {
+        if (isset($age)) {
+            return ($age >= 18 && $age <= 60);
+        } else {
+            return ($this->age >= 18 && $this->age <= 60);
+        }
+
+    }
+    /**
+     * @param int $age
+     * 
+     * @return User
+     */
+    public function setAge(int $age) : User 
+    {
+        if ($this->isAgeCorrect($age)) {
+            $this->age = $age;
+        }
+        return $this;
```

Comment:
Why do you use return in setter?

Reply:
It is made to make chaincall of methods. Without returning the object chaincalls won't work. Example: echo $obj->setAge(20)->getAge();